src/filesystem/nodes/metadata/mod.rs

```diff
@@ -0,0 +1,41 @@
+use std::str::FromStr;
+mod mime_type;
+
+pub use mime_type::MimeGuesser;
+
+#[derive(Hash, Eq, PartialEq)]
+pub enum MetadataKey {
+    MimeType,
+}
```

Comment:
I was under the impression that being able to store arbitrary bytes with an arbitrary key was an intentional design choice to maximize flexibility, but that might be inaccurate. But obviously restricting the key type to an enum has value in terms of organization and consistency.

@sstelfox  any thoughts?

Reply:
Thought the same, but then realized Node is not exposed to the client (if I remember)

Reply:
Metadata entries are going to need to support arbitrary keys. We intend to allow users to set and access any key by string (case-sensitive, string limit of 255 characters) it just isn't fully exposed out to the public facing APIs yet (just on the Node type). There are going to be some "reserved" keywords for things like this.

I do like the idea of using enums for the reserved type, maybe we could just add an extra variant like "User" or "Custom" that contains a String if it doesn't match a reserved type.